hub/saml-engine/src/main/java/uk/gov/ida/hub/samlengine/logging/UnknownMethodAlgorithmLogger.java

```diff
@@ -37,98 +37,73 @@
-            );
-        }
-        else if (!validDigestAlgorithm) {
-            LOG.info(
-                    String.format(DIGEST_ALGORITHM_MESSAGE, role, digestMethodAlgorithm, location)
-            );
+            LOG.info(format(SIGNATURE_AND_DIGEST_ALGORITHMS_MESSAGE, role, signatureMethodAlgorithm, digestMethodAlgorithm, location));
+        } else if (!validSignatureAlgorithm) {
+            LOG.info(format(SIGNATURE_ALGORITHM_MESSAGE, role, signatureMethodAlgorithm, location));
+        } else if (!validDigestAlgorithm) {
+            LOG.info(format(DIGEST_ALGORITHM_MESSAGE, role, digestMethodAlgorithm, location));
         }
     }
 
     public static void probeResponseForMethodAlgorithm(final InboundResponseFromIdp response) {
         if (response != null) {
-            final Optional<Signature> signature = response.getSignature();
-            if (signature != null) {
-                final String signatureMethodAlgorithm = getSignatureMethodAlgorithm(signature);
-                final String digestMethodAlgorithm = getDigestMethodAlgorithm(signature);
-                logMethodAlgorithm(Role.IDP, signatureMethodAlgorithm, digestMethodAlgorithm, Response.DEFAULT_ELEMENT_LOCAL_NAME);
-            }
+            response.getSignature().ifPresent((signature) ->
+                logMethodAlgorithm(Role.IDP, signature, Response.DEFAULT_ELEMENT_LOCAL_NAME)
+            );
         }
     }
 
     public static void probeAssertionForMethodAlgorithm(final Assertion assertion, final String typeOfAssertion) {
         String prefixAssertion = typeOfAssertion + Assertion.DEFAULT_ELEMENT_LOCAL_NAME;
         if (assertion != null) {
-            final Optional<Signature> signature = Optional.ofNullable(assertion.getSignature());
+            Signature signature = assertion.getSignature();
             if (signature != null) {
-                final String signatureMethodAlgorithm = getSignatureMethodAlgorithm(signature);
-                final String digestMethodAlgorithm = getDigestMethodAlgorithm(signature);
-                logMethodAlgorithm(Role.IDP, signatureMethodAlgorithm, digestMethodAlgorithm, prefixAssertion);
+                logMethodAlgorithm(Role.IDP, signature, prefixAssertion);
             }
         }
     }
 
     public static void probeAuthnRequestForMethodAlgorithm(final AuthnRequestFromRelyingParty authnRequest) {
         if (authnRequest != null) {
-            final Optional<Signature> signature = authnRequest.getSignature();
-            if (signature != null) {
-                final String signatureMethodAlgorithm = getSignatureMethodAlgorithm(signature);
-                final String digestMethodAlgorithm = getDigestMethodAlgorithm(signature);
-                logMethodAlgorithm(Role.SP, signatureMethodAlgorithm, digestMethodAlgorithm, AuthnRequest.DEFAULT_ELEMENT_LOCAL_NAME);
-            }
+            authnRequest.getSignature().ifPresent((signature) ->
```

Comment:
Maybe this could throw an NPE now :(

Reply:
as above, a `SamlTransformationErrorException` would be thrown if signature was invalid